File: main/src/main/java/org/anibyl/slounik/core/Versioned.java
package org.anibyl.slounik.core;

import android.os.Build;

/**
 * Contains Android version related stuff.
 * <p/>
 * Created by Usievaład Čorny on 17.07.2016.
 */
public class Versioned {
    public static final String UTF_8 = Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT
            ? java.nio.charset.StandardCharsets.UTF_8.name()
            : "UTF-8";
}

File: main/src/main/java/org/anibyl/slounik/core/Versioned.kt
package org.anibyl.slounik.core

import android.os.Build.VERSION.SDK_INT

/**
 * Contains Android version related stuff.
 *
 * @author Usievaład Kimajeŭ
 * @created 17.07.2016
 */
fun version(version: Int, old: () -> Unit, new: () -> Unit) {
	if (SDK_INT < version) {
		old()
	} else {
		new()
	}
}

fun <T> versionResult(version: Int, old: () -> T, new: () -> T): T {
	if (SDK_INT < version) {
		return old()
	} else {
		return new()
	}
}
